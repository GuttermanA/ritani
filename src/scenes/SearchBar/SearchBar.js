import React, { Component } from "react";
import { Container } from "components";
import { UserSearchForm } from "./components";

class SearchBar extends Component {
  render() {
    const { disabled, handleChange, handleSubmit, username } = this.props;
    return (
      <Container
        className="menu fluid center container"
        data-testid="search-bar-container"
      >
        <UserSearchForm
          username={username}
          handleSubmit={handleSubmit}
          handleChange={handleChange}
          disabled={disabled}
        />
        <a
          style={{ color: "white" }}
          href="https://github.com/GuttermanA/ritani"
          target="_blank"
          rel="noopener noreferrer"
          data-testid="github-repo-link"
        >
          Github Repository
        </a>
      </Container>
    );
  }
}

export default SearchBar;
