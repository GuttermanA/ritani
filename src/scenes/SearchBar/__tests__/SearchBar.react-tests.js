import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import SearchBar from "../SearchBar";

describe("<SearchBar/>", () => {
  afterEach(cleanup);
  it("renders correctly", () => {
    const component = render(<SearchBar />);
    expect(component).toMatchSnapshot();
  });

  it("has the UserSearchForm", () => {
    const { queryByTestId } = render(<SearchBar />);
    expect(queryByTestId("search-form")).toBeTruthy();
  });

  it("has the a link to the Github repo", () => {
    const { queryByTestId } = render(<SearchBar />);
    const link = queryByTestId("github-repo-link");
    expect(link).toBeTruthy();
    expect(link).toHaveAttribute("href");
    expect(link.href).toBe("https://github.com/GuttermanA/ritani");
    expect(link).toHaveAttribute("target");
    expect(link.target).toBe("_blank");
    expect(link).toHaveAttribute("rel");
    expect(link.rel).toBe("noopener noreferrer");
    expect(link).toHaveTextContent("Github Repository");
  });
});
