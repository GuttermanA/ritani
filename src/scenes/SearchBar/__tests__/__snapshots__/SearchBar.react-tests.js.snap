// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar/> renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="menu fluid center container"
        data-testid="search-bar-container"
      >
        <form
          data-testid="search-form"
        >
          <label
            class="form label"
            data-testid="search-label"
            for="username"
          >
            Enter Github Username to Retrieve Followers
          </label>
          <input
            class="form field"
            data-testid="search-field"
            name="username"
            placeholder="Github Username"
            required=""
            type="text"
            value=""
          />
          <button
            class="form button"
            data-testid="search-button"
            type="submit"
          >
            Search
          </button>
        </form>
        <a
          data-testid="github-repo-link"
          href="https://github.com/GuttermanA/ritani"
          rel="noopener noreferrer"
          style="color: white;"
          target="_blank"
        >
          Github Repository
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="menu fluid center container"
      data-testid="search-bar-container"
    >
      <form
        data-testid="search-form"
      >
        <label
          class="form label"
          data-testid="search-label"
          for="username"
        >
          Enter Github Username to Retrieve Followers
        </label>
        <input
          class="form field"
          data-testid="search-field"
          name="username"
          placeholder="Github Username"
          required=""
          type="text"
          value=""
        />
        <button
          class="form button"
          data-testid="search-button"
          type="submit"
        >
          Search
        </button>
      </form>
      <a
        data-testid="github-repo-link"
        href="https://github.com/GuttermanA/ritani"
        rel="noopener noreferrer"
        style="color: white;"
        target="_blank"
      >
        Github Repository
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
